components/link_tooltip/link_tooltip.tsx

```diff
@@ -2,51 +2,65 @@
 // See LICENSE.txt for license information.
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, {RefObject, MouseEvent, CSSProperties} from 'react';
 import Popper from 'popper.js';
 import ReactDOM from 'react-dom';
 
 import {Constants} from 'utils/constants';
 import Pluggable from 'plugins/pluggable';
 
-const tooltipContainerStyles = {
+const tooltipContainerStyles: CSSProperties = {
     display: 'flex',
     flexDirection: 'column',
     alignItems: 'center',
-    zIndex: '10',
+    zIndex: 10,
 };
 
-export default class LinkTooltip extends React.PureComponent {
-    static propTypes = {
+type Props = {
+    href: string;
+    title: string;
+}
+
+export default class LinkTooltip extends React.PureComponent<Props> {
+    public static propTypes = {
         href: PropTypes.string.isRequired,
         title: PropTypes.string.isRequired,
     };
+    private tooltipContainerRef: RefObject<any>;
+    private show: boolean;
+    private hideTimeout: number;
+    private showTimeout: number;
+    private popper?: Popper;
 
-    constructor(props) {
+    public constructor(props: Props) {
         super(props);
 
         this.tooltipContainerRef = React.createRef();
         this.show = false;
+        this.showTimeout = -1;
+        this.hideTimeout = -1;
     }
 
-    showTooltip = (e) => {
+    public showTooltip = (e: any): void => {
         //clear the hideTimeout in the case when the cursor is moved from a tooltipContainer child to the link
-        clearTimeout(this.hideTimeout);
+        window.clearTimeout(this.hideTimeout);
 
         if (!this.show) {
-            const target = $(e.target);
-            const tooltipContainer = $(this.tooltipContainerRef.current);
+            const $target: JQuery = $(e.target);
+            const target: Element = $target.get(0);
+            const $tooltipContainer: JQuery = $(this.tooltipContainerRef.current);
+            const tooltipContainer: any = $tooltipContainer.get(0);
```

Comment:
This breaks with `Uncaught TypeError: tooltipContainer.show is not a function`.  Please try link tooltip with [demo plugin](https://github.com/mattermost/mattermost-plugin-demo).

Reply:
That's a silly error on my part, I used the `Element` rather than the `JQuery` object for the `show()` function. Should be fixed once my changes come through.